### PR TITLE
[GSoC 2023]Student Blog for FASER Real Time Compression Project - SumalyoDatta

### PR DESCRIPTION
**Pull request by Sumalyo**
My student blog for GSoC 2023 
Project - Real-time lossless data compression for the FASER experiment
Mentors: Claire Antel and Brian Petersen

@cantel, Could you please review?

### CODE REVIEW

File: _gsocblogs/2023/blog_FASER_SumalyoDatta.md [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+
+FASER has a trigger-based data acquisition model, and the DAQ software is developed using the DAQling framework.
+>> DAQling is an open-source lightweight C++ software framework that can be used as the core of data acquisition systems of small and medium-sized experiments. It provides a modular system in which custom applications can be plugged in. It also provides the communication layer based on the widespread ZeroMQ messaging library, error logging using ERS, configuration management based on the JSON format, control of distributed applications and extendable operational monitoring with web-based visualization.
+
+You can read more about the strategy in the [paper published by the FASER collaboration](https://arxiv.org/abs/2110.15186).
```
Comment (cantel) [inline _gsocblogs/2023/blog_FASER_SumalyoDatta.md:26]: I would replace "strategy" with "FASER TDAQ design".

File: _gsocblogs/2023/blog_FASER_SumalyoDatta.md [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+
+FASER has a trigger-based data acquisition model, and the DAQ software is developed using the DAQling framework.
+>> DAQling is an open-source lightweight C++ software framework that can be used as the core of data acquisition systems of small and medium-sized experiments. It provides a modular system in which custom applications can be plugged in. It also provides the communication layer based on the widespread ZeroMQ messaging library, error logging using ERS, configuration management based on the JSON format, control of distributed applications and extendable operational monitoring with web-based visualization.
```
Comment (cantel) [inline _gsocblogs/2023/blog_FASER_SumalyoDatta.md:24]: If this is quoted from somewhere, maybe add a link to the document as reference?

File: _gsocblogs/2023/blog_FASER_SumalyoDatta.md [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+
+### The Best Candidate
+| ![Chart](https://sumalyo.github.io/benchmarkFASER/FASERImages/FASER-Report-Compression.jpg) | 
+|:--:| 
+| *Fig 2 Average Compression Ratio vs Average Compression Speed* |
+
+The approach for finding the best compressor is described below
+- Events were divided into a set of 10 classes based on event size (Class 0 having events of smallest size)
+- For each event class, the average compression speed and compression ratio were calculated (for each compressor)
+- The resulting points were plotted on a graph, and this helped to visualize the tradeoff. The compressor configuration offering the highest average compression ratio at the highest compression speed was considered optimal.
```
Comment (cantel) [inline _gsocblogs/2023/blog_FASER_SumalyoDatta.md:70]: A bit nitpicky since I know what you mean: Since there is a trade-off between compression ratio and speed, we will not get an algorithm that will achieve _both_ highest compression _and_ highest speed. I would rephrase by referring to your "acceptance region" in fig 2 and something like "The compressor configuration configuration offering the highest compression speed at an acceptable compression ratio" (or vice versa).

**Comment by cantel** [comment]
> My student blog for GSoC 2023 Project - Real-time lossless data compression for the FASER experiment Mentors: Claire Antel and Brian Petersen
> 
> @cantel, Could you please review?

very nice, thanks :)

**Comment by vvolkl** [comment]
@cantel Seems like your comments were not followed up on, should we merge anyway or close?

File: _gsocblogs/2023/blog_FASER_SumalyoDatta.md [hunk 4]
```diff
@@ -67,7 +73,7 @@ The logs and the python-based analysis notebooks can be found at [this repo](htt
 The approach for finding the best compressor is described below
 - Events were divided into a set of 10 classes based on event size (Class 0 having events of smallest size)
 - For each event class, the average compression speed and compression ratio were calculated (for each compressor)
-- The resulting points were plotted on a graph, and this helped to visualize the tradeoff. The compressor configuration offering the highest average compression ratio at the highest compression speed was considered optimal.
+- The resulting points were plotted on a graph, and this helped to visualize the tradeoff. A compression ration of about 2 (or 50% compression ) and a speed more that 40 MB/s was considered as acceptable performance (as denoted in the diagram above).The compressor configuration offering the highest average compression ratio at the highest compression speed (in the acceptable performance region of the graph) was considered optimal.
```
Comment (cantel) [inline _gsocblogs/2023/blog_FASER_SumalyoDatta.md:76]: Many thanks, @Sumalyo . Since it's such a nice blog entry, I'm adding one last comment on your last commit as it includes some grammar mistakes: "compression ration" ->"compression ratio", "and a speed more that 40 MB/s" -> "and a compression speed of more than 40 MB/s"

**Comment by cantel** [comment]
> @cantel Seems like your comments were not followed up on, should we merge anyway or close?

Hi @vvolkl , thanks for the nudge and thanks @Sumalyo  for the quick response. Sorry, I should have follow up much earlier as well.

File: _gsocblogs/2023/blog_FASER_SumalyoDatta.md (hunk 4 above)
Comment (Sumalyo) [reply _gsocblogs/2023/blog_FASER_SumalyoDatta.md]: Hi Claire,
I have made the changes that you have suggested

**Comment by Sumalyo** [comment]
Thanks @cantel for your feedback; thank you @vvolkl for reminding me, I am sorry again for the delay.

**Review by vvolkl — APPROVED** [review]
